include/aspect/simulator.h

```diff
@@ -1434,6 +1434,20 @@ namespace aspect
            */
           ConstraintMatrix mesh_vertex_constraints;
 
+          /**
+           * A struct for holding information about how to advect the free surface.
+           */
+          struct SurfaceAdvection
+          {
+            enum Direction { normal, gravity };
+          };
+
+          /**
+           * Stores whether to advect the free surface in the normal direction
+           * or the direction of gravity.
+           */
+          typename SurfaceAdvection::Direction advection_direction;
```

Comment:
Hmm, compilation fails without it.  I modeled this struct after the NonlinearSolver and NullspaceRemoval structs.